core/src/main/java/com/linecorp/armeria/client/HttpRequestSubscriber.java

```diff
@@ -218,7 +242,12 @@ public void onNext(HttpObject o) {
     @Override
     public void onError(Throwable cause) {
         isSubscriptionCompleted = true;
-        failAndRespond(cause);
+        if (id >= 0) { // onSubscribe is called.
+            failAndReset(cause);
+        } else {
+            // No need to send RST because we didn't send any packet.
+            fail(cause);
```

Comment:
Don't we need to wrap this cause with `UnprocessedRequestException?

Reply:
Yes, right. Thanks!